### PR TITLE
switching official build process to gulp (includes protractor testing)

### PR DESCRIPTION
**Pull request by martinraison**
@andrewconner @yingjieMiao @stkem

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/gulpfile.js [hunk 1]
```diff
@@ -349,7 +349,7 @@ gulp.task('karma', ['templates'], function (done) {
 });
 
 gulp.task('protractor', ['build-release'], function () {
-  startProdServer();
+  startProdServer(9080);
```
Comment (martinraison) [inline kifi-backend/modules/shoebox/angular/gulpfile.js:352]: marvin already uses port 8080 so we need to use another one

File: tools/setup_chrome_headless.sh [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+# Optional but nifty: For capturing screenshots of Xvfb display:
+sudo apt-get -y install imagemagick x11-apps
+ 
+# Make sure that Xvfb starts everytime the box/vm is booted:
+echo "Starting X virtual framebuffer (Xvfb) in background..."
+Xvfb -ac :99 -screen 0 1280x1024x16 &
+export DISPLAY=:99
+ 
+# Optionally, capture screenshots using the command:
+#xwd -root -display :99 | convert xwd:- screenshot.png
```
Comment (martinraison) [inline tools/setup_chrome_headless.sh:57]: This script is there so we can remember how to run headless Chrome on Ubuntu.

File: tools/setup_chrome_headless.sh [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+sudo apt-get -y install xvfb gtk2-engines-pixbuf
+sudo apt-get -y install xfonts-cyrillic xfonts-100dpi xfonts-75dpi xfonts-base xfonts-scalable
+ 
+# Optional but nifty: For capturing screenshots of Xvfb display:
+sudo apt-get -y install imagemagick x11-apps
+ 
+# Make sure that Xvfb starts everytime the box/vm is booted:
+echo "Starting X virtual framebuffer (Xvfb) in background..."
+Xvfb -ac :99 -screen 0 1280x1024x16 &
+export DISPLAY=:99
```
Comment (martinraison) [inline tools/setup_chrome_headless.sh:54]: I'm adding these two commands to the jenkins build script (I'm sure there is a way to setup marvin to run them on startup, but we want to make migration easier)
The first command does nothing is Xvfb is already running.

**Comment by yipingliao** [comment]
Awesome! Thanks for figuring out the the xvfb stuff. :)

It looks like "gulp test" still depends on "run-karma-tests" and "run-protractor-tests", which have been renamed; I think you're fixing that with Josh.

**Comment by yipingliao** [comment]
Sorry, one more thing: add sprites.png to .gitignore?

**Comment by martinraison** [comment]
I think we can add sprites.png, sprites.styl and spritesClasses.styl to .gitignore when we remove the gruntfile and grunt dependencies. In the meantime it is probably better to make sure that grunt is still working as expected.

**Comment by 2is10** [comment]
like!

**Pull request merged**